account_general_ledger_report/models/general_ledger_report.py

```diff
@@ -0,0 +1,244 @@
+        period_name = map(lambda x: x[1], self._cr.fetchall())
+        periods = Period.search(
+            [('name', 'in', period_name)], order='name')
+        return (periods, moves)
+
+    @api.model
+    def _get_period_moves(self, report, period):
+        MoveLine = self.env['account.move.line']
+        period_moves = MoveLine.search(
+            [('period_id.fiscalyear_id', '=', report.fiscalyear_id.id),
+             ('period_id', '<', period.id),
+             ('centralisation', '=', 'normal'), ])
+        return period_moves
+
+    @api.model
+    def _get_focus_moves(self, moves, period):
+        MoveLine = self.env['account.move.line']
+        focus_moves = MoveLine.search(
+            [('period_id', '=', period.id),
+             ('centralisation', '=', 'normal'),
+             ('id', 'in', moves.ids)])
+        return focus_moves
+
+    @api.model
+    def generate_report(self, fiscalyear_id, target_move, reconcile_cond,
+                        account_ids, amount_currency):
+        # Delete old reports
+        self.search(
+            [('create_uid', '=', self.env.user.id),
+             ('create_date', '<', fields.Date.context_today(self))]).unlink()
+
+        # Create report
+        name = 'General Ledger (Web)'
+        report = self.create({'name': name,
+                              'fiscalyear_id': fiscalyear_id,
+                              'target_move': target_move,
+                              'reconcile_cond': reconcile_cond,
+                              'account_ids': [(6, 0, account_ids.ids)],
+                              'amount_currency': amount_currency})
+
+        # Compute report lines
+        periods, moves = self._get_moves(fiscalyear_id, target_move,
+                                         reconcile_cond, account_ids,
+                                         amount_currency)
+
+        report_lines = []
+        accum = 0.0
+
+        for period in periods:
+            focus_moves = self._get_focus_moves(moves, period)
+            debit = 0.0
+            credit = 0.0
+            # Focus
+            if focus_moves:
+                self._cr.execute("""
+                    select coalesce(sum(debit), 0.0) debit,
+                        coalesce(sum(credit), 0.0) credit
+                    from account_move_line where id in %s
+                """, (tuple(focus_moves.ids),))
+                res = self._cr.fetchone()
+                (debit, credit) = (res[0], res[1])
+
+            line_dict = {
+                'period_id': period.id,
+                'debit': debit,
+                'credit': credit,
+                'balance': debit - credit,
+                'accum_balance': accum + (debit - credit),
+            }
+            accum = line_dict.get('accum_balance')
+            report_lines.append((0, 0, line_dict))
+        report.write({'line_ids': report_lines})
+        return report.id
+
+    @api.model
+    def vacumm_old_reports(self):
```

Comment:
Have you create Job for this method?

This method should be called by scheduled job to delete old reports.

Reply:
You can search for external id -> cronjob_budget_drilldown_report_autovacuum
to see sample.

Reply:
Note: Do not leave any code that is not called by anyone. (in doubt, please consult).